indy_node/server/request_handlers/domain_req_handlers/rs_schema_handler.py

```diff
@@ -0,0 +1,80 @@
+from indy_common.authorize.auth_actions import AuthActionAdd, AuthActionEdit
+from indy_common.authorize.auth_request_validator import WriteRequestValidator
+
+from indy_common.constants import SET_RS_SCHEMA, \
+    RS_META, RS_META_VERSION, RS_DATA, RS_META_NAME, DOMAIN_LEDGER_ID
+
+from indy_common.state.state_constants import MARKER_RS_SCHEMA
+
+from plenum.common.request import Request
+from plenum.server.database_manager import DatabaseManager
+from plenum.server.request_handlers.handler_interfaces.write_request_handler import WriteRequestHandler
+from plenum.server.request_handlers.utils import encode_state_value
+
+
+class RsSchemaHandler(WriteRequestHandler):
+    def __init__(self, database_manager: DatabaseManager,
+                 write_req_validator: WriteRequestValidator):
+        super().__init__(database_manager, SET_RS_SCHEMA, DOMAIN_LEDGER_ID)
+        self.write_req_validator = write_req_validator
+
+    def static_validation(self, request: Request):
+        pass
+
+    @staticmethod
+    def validate_schema(schema):
+        pass
+
+    def dynamic_validation(self, request: Request):
+        author = request.identifier
+        name = request.operation[RS_META][RS_META_NAME]
+        version = request.operation[RS_META][RS_META_VERSION]
+        state_path = RsSchemaHandler.make_state_path(author, name, version).decode("utf-8")
+        schema, _, _ = self.get_from_state(state_path)
+        if schema:
+            self.write_req_validator.validate(request,
+                                              [AuthActionEdit(txn_type=SET_RS_SCHEMA,
+                                                              field='*',
+                                                              old_value='*',
+                                                              new_value='*')])
+        else:
+            self.write_req_validator.validate(request,
+                                              [AuthActionAdd(txn_type=SET_RS_SCHEMA,
+                                                             field='*',
+                                                             value='*')])
+
+    def gen_txn_id(self, txn):
+        self._validate_txn_type(txn)
+        path = RsSchemaHandler.prepare_state(txn, path_only=True)
+        return path.decode()
+
+    def update_state(self, txn, prev_result, request, is_committed=False) -> None:
+        self._validate_txn_type(txn)
+        path, value_bytes = RsSchemaHandler.prepare_state(txn)
+        self.state.set(path, value_bytes)
+
+    @staticmethod
+    def prepare_state(txn, path_only=False):
+        payload = txn["txn"]
+        meta, data = payload[RS_META], payload[RS_DATA]
+        did_author = txn["txn"]["metadata"].get("from", None)  # why not txn.identifier
```

Comment:
`get_from` helper method can be used here instead

Reply:
My original reasoning for not using `get_from` was to keep dependencies down and explicit readability. I still favor `txn["txn"]["metadata"].get("from", None)` over `get_from`, do I need to change it?